File: frontend/src/config/feature-flags.ts
export type FeatureTier = 'free' | 'pro'

export const FEATURE_FLAGS: Record<string, FeatureTier> = {
  theme_intelligence: 'free',
  options_chain: 'free',
  insider_trades: 'free',
  sentiment_analysis: 'free',
  trend_reversal: 'free',
  onchain_prediction: 'free',
} as const

export type FeatureName = keyof typeof FEATURE_FLAGS

export function getRequiredTier(feature: FeatureName): FeatureTier {
  return FEATURE_FLAGS[feature] ?? 'pro'
}

export function isFeatureAvailable(feature: FeatureName, userTier: FeatureTier): boolean {
  const required = getRequiredTier(feature)
  if (required === 'free') return true
  return userTier === 'pro'
}

File: backend/app/core/feature_flags.py
"""
Central feature flag config.
Change "free" → "pro" for any feature to gate it behind a Pro subscription.
No other code changes needed — require_feature() and ProGate read from here.
"""

FEATURE_FLAGS: dict[str, str] = {
    "theme_intelligence":  "free",
    "options_chain":       "free",
    "insider_trades":      "free",
    "sentiment_analysis":  "free",
    "trend_reversal":      "free",
    "onchain_prediction":  "free",
}


def get_required_tier(feature: str) -> str:
    return FEATURE_FLAGS.get(feature, "pro")
